lib/linguist/generated.rb

```diff
@@ -301,7 +301,11 @@ def generated_go?
       return false unless extname == '.go'
       return false unless lines.count > 1
 
-      return lines[0].include?("Code generated by")
+      # old check for the transitional period (remove it later)
+      return true if lines[0].include?("Code generated by")
+
+      # https://golang.org/s/generatedcode
+      return lines.find {|line| line =~ /^\/\/ Code generated .* DO NOT EDIT\.$/ }
```

Comment:
I understand the rule states this could be anywhere in the file, but in practice, isn't it almost always in the first few lines? I think we should avoid scanning entire files if possible.

Reply:
As far as I could find the very first (comment) lines would contain it, have seen them in 17th line too (due to long (generated) description text. I agree checking files with 10k and more lines is too much.

What would be a solution, collecting first consecutive comment lines and testing them?

Reply:
> What would be a solution, collecting first consecutive comment lines and testing them?

Please check the other examples that match against several lines below (e.g., `lines.first(6).`).

> As far as I could find the very first (comment) lines would contain it, have seen them in 17th line too due to long (generated) description text. I agree checking files with 10k and more lines is too much.

If the 17 lines case happens once in a lifetime, it's probably not worth it and we can match only on say the first 10 lines. If you've seen several such cases, then let's go for 17 lines :-)

Reply:
It looks like the original intent was to match on go-bindata generated files (cf. #2426), which won't match the second rule. So let's keep the first rule!

Reply:
Good points, updated the pull request to test first 10 lines